lib/orbit-common/source.js

```diff
@@ -255,6 +314,35 @@ var Source = Class.extend({
   // Internals
   /////////////////////////////////////////////////////////////////////////////
 
+  _normalizeId: function(type, id) {
+    if (isObject(id)) {
+      var record = this.normalize(type, id);
+      id = this.getId(type, record);
+    }
+    return id;
+  },
+
+  _normalizeLink: function(type, key, value) {
+    if (isObject(value)) {
+      var linkDef = this.schema.modelDefinition(type).links[key];
+      var relatedRecord;
+
+      if (isArray(value)) {
+        for (var i = 0, l = value.length; i < l; i++) {
```

Comment:
Do we want to edit this array in place?  Would it be be better to `value = value.map` here?